src/main/java/com/google/step/snippet/external/GeeksForGeeksClient.java

```diff
@@ -51,9 +56,28 @@ public Card search(String geeksLink) {
     if (snippets.isEmpty() || snippets.first().getElementsByClass(CODE_CLASS).text().isEmpty()) {
       return null;
     }
-    String title = titles.first().text();
-    String description = descriptions.first().text();
+    String title = Jsoup.clean(titles.first().text(), Whitelist.relaxed());
+    String description = Jsoup.clean(descriptions.first().text(), Whitelist.relaxed());
     String code = snippets.first().getElementsByClass(CODE_CLASS).text();
+    if (containsEscape(query.toLowerCase())
+        || containsEscape(geeksLink)
+        || containsEscape(title.toLowerCase())
+        || containsEscape(description.toLowerCase())) {
+      title = StringEscapeUtils.escapeHtml4(title);
+      description = StringEscapeUtils.escapeHtml4(description);
+      code = StringEscapeUtils.escapeHtml4(code);
+    } else {
+      code = Jsoup.clean(code, Whitelist.relaxed().preserveRelativeLinks(true));
```

Comment:
Why do you want to preserve relative links? Won't any such links fail, since the resource won't be hosted on your app?

Reply:
Sure, but I thought preserving these links might provide useful context in an answer for an html related question.

Reply:
Old code from when I was cleaning all displayed code, I thought preserving links might help provided context to html related questions.